skbeam/core/recip.py

```diff
@@ -342,3 +342,37 @@ def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
     qr = np.sqrt(qx**2 + qy**2)
 
     return gisaxs_output(alpha_i, theta_f, alpha_f, tilt_angle, qx, qy, qz, qr)
+
+
+def generate_q_bins(rmax, qmax, pixel_size, distance, wavelength):
+    """
+    Generate the Q bins at the resolution of the detector
```

Comment:
This generates non uniform-width bins such that each bin size in q space maps back to a single pixel?